### PR TITLE
add support for network enrollment token enrollment

### PR DESCRIPTION
**Pull request by ekoby**
- simplify enrollment code memory usage
- fix memory leaks

### CODE REVIEW

File: library/zitilib.c [hunk 1]
```diff
@@ -1171,9 +1171,9 @@ static void do_enroll(ziti_enroll_opts *opts, future_t *f, uv_loop_t *loop) {
 
 int Ziti_enroll_identity(const char *jwt, const char *key, const char *cert, char **id_json, unsigned long *id_json_len) {
     ziti_enroll_opts opts = {
-            .jwt_content = jwt,
-            .enroll_key = key,
-            .enroll_cert = cert,
+            .token = jwt,
```
Comment (dovholuknf) [inline library/zitilib.c:1174]: why rename these and why would this only be a minor bump? shouldn't this be a major bump since the api is changing?

Comment (ekoby) [reply library/zitilib.c:1174]: `enroll_` prefix is redundant. it is minor to be consistent with how we always versioned things

File: library/ziti_enroll.c [hunk 2]
```diff
@@ -91,210 +140,361 @@ int ziti_enroll(const ziti_enroll_opts *opts, uv_loop_t *loop,
...
-            ZITI_LOG(WARN, "failed to load private key[%s]", enroll_req->ecfg->private_key);
-            if (enroll_req->ecfg->zej.method == ziti_enrollment_methods.ott &&
-                strncmp(enroll_req->ecfg->private_key, "pkcs11://", strlen("pkcs11://")) == 0) {
-                ZITI_LOG(INFO, "attempting to generate pkcs11 key");
-                TRY(TLS, gen_p11_key_internal(tls, &enroll_req->ecfg->pk, enroll_req->ecfg->private_key));
-            }
+    if (url.query != NULL) {
+        // TODO magic URL
+    } else {
+        // controller URL must be verifiable with default/OS cert bundle
```
Comment (dovholuknf) [inline library/ziti_enroll.c:214]: will there be a clear log message or error when trying to use network jwt without a verifiable os / cert bundle?

**Review by dovholuknf — APPROVED** [review]

File: includes/ziti/ziti.h [hunk 3]
```diff
@@ -29,6 +29,7 @@
 #include "ziti_model.h"
 #include "enums.h"
 #include "ziti_events.h"
+#include "enroll.h"
```
Comment (scareything) [inline includes/ziti/ziti.h:32]: "ziti/enroll.h"?

File: includes/ziti/enroll.h [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ZITI_ENROLL_H
+#define ZITI_ENROLL_H
+
+#include <stdbool.h>
+#include "ziti_model.h"
```
[the remark this answers is not in the document]
Comment (ekoby) [reply includes/ziti/enroll.h:19]: it should work "ziti_model.h" will also resolve against this file's dir

**Pull request merged**